### ISSUE TITLE
Parsing error: Unexpected character '#'

### ISSUE THREAD

**Issue by limpid-kzonix** [eslint/eslint#13589]
**Please describe what the rule should do:**
Support [Private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)

**What new ECMAScript feature does this rule relate to?**
[Private class fields](https://github.com/tc39/proposal-class-fields)

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ X ] Other (Don't complain about private class fields with error: ESLint: Parsing error: Unexpected character '#')

**Provide 2-3 code examples that this rule will warn about:**

```js
export class Example {
  #privateClassField

  constructor() {
    this.#privateClassField = ...
  }
 
```

**Why should this rule be included in ESLint (instead of a plugin)?**
This feature already supported in LTS version of node.
![image](https://user-images.githubusercontent.com/8176996/90566444-2980f680-e1b1-11ea-9071-49ec894b9e73.png)

**Are you willing to submit a pull request to implement this rule?**
No

**Comment by mdjermanovic** [comment]
Hi @limpid-kzonix, thanks for the issue!

The class fields proposal is currently in [stage 3](https://github.com/tc39/proposal-class-fields).

Per our [policy about experimental features](https://github.com/eslint/eslint#what-about-experimental-features), ESLint's default parser will support this syntax if and when it reaches stage 4.

In the meantime, you can use ESLint with an [external parser](https://eslint.org/docs/user-guide/configuring#specifying-parser), like [@babel/eslint-parser](https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser) which already supports this syntax.

**Comment by limpid-kzonix** [comment]
Hi @mdjermanovic, Thanks for pointing me to the solutions.

I ran into a little babel-parser issue. And to be exact, the issue with eslint/lib/rules/no-unused-vars, only in case of using babel-parser.

I have prepared a minor fix for that one. But I'm worried the problem could be more in babel parser rather than eslint rule itself.

Could you please check the PR, I have attached a little description of the error there
https://github.com/eslint/eslint/pull/13592
